lib/mayaUsd/ufe/UsdPointInstanceModifierBase.h

```diff
@@ -30,6 +31,28 @@
 namespace MAYAUSD_NS_DEF {
 namespace ufe {
 
+//! \brief Shared data for USD point instance read and write batching.
+//
+// This data structure can be used by UsdPointInstanceModifierBase when
+// modifying the position, orientation, or scale of multiple point instances
+// from a single point instancer.  These changes can be batched such that the
+// first modifier to make a change will read from the point instancer attribute
+// into the usdValues array, and the last modifier to make a change will write
+// the usdValues array back to the point instancer attribute.  All modifiers in
+// the batch will write to and read from the usdValues array, a much less
+// expensive operation than writing to or reading from the point instancer
+// attribute.
+//
+template <class UsdValueType> struct MAYAUSD_CORE_PUBLIC UsdPointInstanceBatch
+{
+    inline bool isReader() const { return (count % nbInstances) == 0; }
+    inline bool isWriter() const { return ((count + 1) % nbInstances) == 0; }
+
+    PXR_NS::VtArray<UsdValueType> usdValues;
+    int                           nbInstances { 0 };
```

Comment:
This can never be negative, consider switching to one of the unsigned types.

Reply:
It would be useful as well to document here how the number of instances is discovered.

Reply:
Because of problems with mixed integral type arithmetic
https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Res-mix
I usually don't use signed integers (John Lakos has a minor design rule on this, unfortunately don't have his book handy).

However, in this case, because I'm using modular arithmetic, I think your point is well taken.